server/node.go

```diff
@@ -426,10 +426,16 @@ func (n *Node) gossipCapacities() {
 // ctx.ScanInterval and store the status in the db.
 func (n *Node) startStoresScanner(stopper *util.Stopper) {
 	stopper.RunWorker(func() {
+		// Pick the smalled of the two intervals.
+		var minScanInterval time.Duration
+		if n.ctx.ScanInterval <= n.ctx.ScanMaxIdleTime {
```

Comment:
I didn't realize maxIdleTime could be 0.  That would be dangerous here.
I'm going to actually have this run every time a store is updated.  I've got a branch where I started this work, but ran into some weird issues and didn't have time to debug them this morning.  
I'll create a quick PR to set the setting to MaxIdleTime only if it's non-zero as a temporary fix.